app/services/xccdf_report_parser.rb

```diff
@@ -67,9 +67,13 @@ def check_for_external_reports
   def check_for_missing_benchmark
     # rubocop:disable Style/GuardClause
     unless benchmark.persisted?
-      raise UnknownBenchmarkError,
-            "No benchmark found matching ref_id #{benchmark.ref_id} and "\
-            "SSG #{benchmark.version}. #{parse_failure_message}"
+      if SupportedSsg.versions.include?(benchmark.version)
+        save_all_benchmark_info
```

Comment:
Observation: Unexpected side-effect on a check method. I'd suggest to move this (with the condition) before doing the group of checks.

Reply:
Updated